### PR TITLE
start removing use of CompressedSparsityPattern and friends

### PR DESCRIPTION
**Pull request by guidokanschat**
Stop using CompressedSparsityPattern and friends in the library after they have been deprecated

### CODE REVIEW

File: include/deal.II/dofs/dof_tools.h [hunk 1]
```diff
@@ -362,9 +362,9 @@ namespace DoFTools
    * Remember using SparsityPattern::compress() after generating the pattern.
    *
    * The actual type of the sparsity pattern may be SparsityPattern,
-   * CompressedSparsityPattern, BlockSparsityPattern,
-   * BlockCompressedSparsityPattern, BlockCompressedSetSparsityPattern,
-   * BlockCompressedSimpleSparsityPattern, or any other class that satisfies
+   * DynamicSparsityPattern, BlockSparsityPattern,
+   * BlockDynamicSparsityPattern, BlockDynamicSetSparsityPattern,
+   * BlockDynamicSimpleSparsityPattern, or any other class that satisfies
```
Comment (bangerth) [inline include/deal.II/dofs/dof_tools.h]: The last two of these don't actually exist -- only `DynamicSparsityPattern` is left, no `*Set*` and `*Simple*` variants any more. Same further down below.

File: include/deal.II/lac/sparse_matrix.templates.h [hunk 2]
```diff
@@ -967,9 +967,9 @@ SparseMatrix<number>::mmult (SparseMatrix<numberC>       &C,
       // rows in the matrix A, and for each column in a row we add the whole
       // row of matrix B with that row number. This means that we will insert
       // a lot of entries to each row, which is best handled by the
-      // CompressedSimpleSparsityPattern class.
+      // DynamicSparsityPattern class.
       {
-        CompressedSimpleSparsityPattern csp (m(), B.n());
+        DynamicSparsityPattern csp (m(), B.n());
```
Comment (bangerth) [inline include/deal.II/lac/sparse_matrix.templates.h]: I think we should make the same `csp->dsp` substitution here as we did in the tutorial programs.

File: source/grid/grid_tools.cc [hunk 3]
```diff
@@ -693,7 +692,7 @@ namespace GridTools
     DoFHandler<dim> dof_handler(triangulation);
     dof_handler.distribute_dofs(q1);
 
-    CompressedSparsityPattern c_sparsity_pattern (dof_handler.n_dofs (),
+    DynamicSparsityPattern c_sparsity_pattern (dof_handler.n_dofs (),
                                                   dof_handler.n_dofs ());
```
Comment (bangerth) [inline source/grid/grid_tools.cc]: Indentation. You probably need to let the indentation script run over everything.

**Comment by bangerth** [comment]
Nice work, thanks. I've seen no other things than the ones I marked up.

File: include/deal.II/lac/sparse_matrix.templates.h (hunk 2 above)
Comment (tjhei) [inline include/deal.II/lac/sparse_matrix.templates.h]: yes, I decided to use ```dsp``` everywhere except in the very first tutorial.

**Comment by tjhei** [comment]
Thanks for looking into this, Guido. Looking good. :+1:

**Comment by guidokanschat** [comment]
Replaced all the occurrences of csp and c_sparsity. Will have to run tests later.

Currently, I have not worked through the test suite

**Comment by bangerth** [comment]
Good work, thanks!

**Pull request merged**